UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetGenerateFilesForInValidProductVersion.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Newtonsoft.Json;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helper;
+using UKHO.ExchangeSetService.API.FunctionalTests.Models;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    class ExchangeSetGenerateFilesForInValidProductVersion
+    {
+        private string EssJwtToken { get; set; }
+        private string FssJwtToken { get; set; }
+        private ExchangeSetApiClient ExchangeSetApiClient { get; set; }
+        private FssApiClient FssApiClient { get; set; }
+        private TestConfiguration Config { get; set; }
+        public DataHelper DataHelper { get; set; }
+        private SalesCatalogueApiClient ScsApiClient { get; set; }
+        private string ScsJwtToken { get; set; }
+        private string DownloadedFolderPath { get; set; }
+        private List<ProductVersionModel> ProductVersiondata { get; set; }
+
+        [OneTimeSetUp]
+        public async Task SetupAsync()
+        {
+            Config = new TestConfiguration();
+            ExchangeSetApiClient = new ExchangeSetApiClient(Config.EssBaseAddress);
+            FssApiClient = new FssApiClient();
+            AuthTokenProvider authTokenProvider = new AuthTokenProvider();
+            EssJwtToken = await authTokenProvider.GetEssToken();
+            FssJwtToken = await authTokenProvider.GetFssToken();
+            DataHelper = new DataHelper();
+            ScsApiClient = new SalesCatalogueApiClient(Config.ScsAuthConfig.ScsApiUrl);
+            ScsJwtToken = await authTokenProvider.GetScsToken();
+            ProductVersiondata = new List<ProductVersionModel>();////Invalid Edition Number
+            ProductVersiondata.Add(DataHelper.GetProductVersionModelData("DE416080", 20, 5)); 
+            DownloadedFolderPath = await ValidateGeneratedFile();
+        }
+
+        public async Task<string> ValidateGeneratedFile()
+        {
+            var apiEssResponse = await ExchangeSetApiClient.GetProductVersionsAsync(ProductVersiondata, accessToken: EssJwtToken);
+            Assert.AreEqual(200, (int)apiEssResponse.StatusCode, $"Incorrect status code is returned {apiEssResponse.StatusCode}, instead of the expected status 200.");
+
+            var apiResponseData = await apiEssResponse.ReadAsTypeAsync<ExchangeSetResponseModel>();
+
+            var batchStatusUrl = apiResponseData.Links.ExchangeSetBatchStatusUri.Href;
+
+            var batchStatus = await FssBatchHelper.CheckBatchIsCommitted(batchStatusUrl.ToString(), FssJwtToken);
+            Assert.AreEqual("Committed", batchStatus, $"Incorrect batch status is returned {batchStatus}, instead of the expected status is Committed.");
```

Comment:
Kindly remove "is" from the statement "instead of the expected status is Committed". Do this change in all 3 test case files.

Reply:
Resolved